### PR TITLE
PR: Fix/Find in files

### PR DESCRIPTION
**Pull request by goanpeca**

### CODE REVIEW

File: spyderlib/config/utils.py [hunk 1]
```diff
@@ -80,15 +80,23 @@ def _get_pygments_extensions():
...
+            # Reference: This commented line was replaced by the following one
+            # to trim only extensions that start with '*'
+            # lexer_exts = [le[1:] for le in lexer_exts]
+            lexer_exts = [le[1:] for le in lexer_exts if le.startswith('*')]
             lexer_exts = [le for le in lexer_exts if not le.endswith('_*')]
-            extensions = extensions + list(lexer_exts)
-    return extensions
+            extensions = extensions + list(lexer_exts) + list(other_exts)
+
+    return sorted(list(set(extensions)))
```
Comment (ccordoba12) [inline spyderlib/config/utils.py]: What do these changes do exactly?

File: spyderlib/plugins/findinfiles.py [hunk 2]
```diff
@@ -94,14 +97,23 @@ def findinfiles_callback(self):
...
+        patterns = [r'|'.join([ext + r'$' for ext in clean_exts if ext]) +
                     r'|README|INSTALL',
-                    r'\.pyw?$|\.ipy$|\.txt$|\.rst$',
-                    '.']
+                    r'\.ipy$|\.pyw?$|\.rst$|\.txt$',
+                    '.',
+                    ]
         return patterns
-    
+
```
Comment (ccordoba12) [inline spyderlib/plugins/findinfiles.py]: Please remove this blank :-)

**Comment by ccordoba12** [comment]
Does your changes hide the default (advanced) view of *Find in files*? If not, I'd like to see that as part of this PR ;-)

**Comment by goanpeca** [comment]
Hi @ccordoba12 , I forgot to add the description. Basically the include regex where based on the extension provided by the pygments lexers, but the problem is that the generated regex was invalid. I corrected that and added some tests so we are sure that the Include and Exclude patterns are always valid regular expressions.

I can hide the pane by default. Working on a ack/grin like interface after this PR.

File: spyderlib/config/utils.py (hunk 1 above)
Comment (goanpeca) [inline spyderlib/config/utils.py]: `lexer_exts = [le[1:] for le in lexer_exts]`

This code is wrong cause not all lexers extensions start with a **`*`**, so I fixed that so the actual generated patterns are valid regex. (this was causing find in files to not work after a clean install)

**Comment by goanpeca** [comment]
@ccordoba12, now the advanced options are hidden by default

**Comment by ccordoba12** [comment]
I tested our edit filters and they are working correctly, so merging ;-)

**Pull request merged**